tasks.yaml

```diff
@@ -0,0 +1,63 @@
+variables:
+  - name: MODULES
+    default: $(find . -mindepth 2 -maxdepth 4 -type f -name 'go.mod' | cut -c 3- | sed 's|/[^/]*$$||' | sort -u | tr / :)
+
+
+tasks:
+  - name: set-modules-variable
+    actions:
+      - cmd: echo $(find . -type f -name 'go.mod' -exec dirname {} \; | cut -c 3-)
+        setVariables:
+          - name: MODULES
+
+  # This could be improved if something like this was introduced https://github.com/defenseunicorns/maru-runner/issues/17
+  - name: build-pkg
+    description: build specified package
+    actions:
+      - cmd: cd "${PKG}" &&  go build -o ../build/${PKG} .
+        description: "build pkg"
+
+  - name: build-all
+    description: build all packages
+    actions:
+      - task: set-modules-variable
+      - cmd: |
+          for dir in ${MODULES}; do
+            uds run build-pkg --set PKG=$dir
+          done
+        description: build all
```

Comment:
`revive` + `go test` is going to catch compiler errors, no need to build the modules.

Reply:
I added it so we can do a build for codeql

Reply:
Played around with using `go vet` or `go test`, but codeQL will still fail on both